Providers/Scripts/2.6x-2.7x/Scripts/nxOMSPerfCounter.py

```diff
@@ -157,14 +173,12 @@ def TranslatePerfs(object_name, perfs):
 
 
 def ReadOMSAgentConf(HeartbeatIntervalSeconds, PerfCounterObject):
-    txt = ''
-    try:
-        txt = codecs.open(conf_path, 'r', 'utf8').read().encode(
-            'ascii', 'ignore')
-        LG().Log('INFO', 'Read omsagent configuration ' + conf_path + '.')
-    except:
-        LG().Log(
-            'ERROR', 'Unable to read omsagent configuration ' + conf_path + '.')
+    """
+    Read OMSAgent conf file and extract the current settings for
+    HeartbeatIntervalSeconds and perf objects
+    """
+    txt = get_conf_path_text()
+
     heartbeat_srch_str = r'<source>.*?tag heartbeat.*?run_interval ([0-9]+[a-z])\n</source>\n'
```

Comment:
Does not take into account txt is an empty string case?

Reply:
It never explicitly took that case into account; if txt is an empty string, then all of the searches will return no results, and `new_heartbeat` will be returned as `None`, and `new_perfobj` will be an empty array.

Reply:
Yes, then why even do the searches if they are going to return no results?

Reply:
I've added a check to exit early if there is no txt to search.